src/v/config/rest_authn_endpoint.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "config/rest_authn_endpoint.h"
+
+#include "kafka/client/exceptions.h"
+#include "model/metadata.h"
+#include "utils/string_switch.h"
+
+namespace config {
+
+constexpr inline std::string_view to_string_view(rest_authn_type m) {
+    switch (m) {
+    case rest_authn_type::none:
+        return "none";
+    case rest_authn_type::http_basic:
+        return "http_basic";
+    default:
+        return "invalid";
+    }
+}
+
+template<>
+std::optional<rest_authn_type>
+from_string_view<rest_authn_type>(std::string_view sv) {
+    return string_switch<std::optional<rest_authn_type>>(sv)
+      .match("none", rest_authn_type::none)
+      .match("http_basic", rest_authn_type::http_basic)
+      .default_match(std::nullopt);
+}
+
+std::ostream& operator<<(std::ostream& os, const rest_authn_endpoint& ep) {
+    fmt::print(os, "{{{}:{}:{}}}", ep.name, ep.address, ep.authn_type);
```

Comment:
Does this print the name of the auth_type or its integral value? I think name would be preferred.

Reply:
It prints the name.

Reply:
Hmm. I was expecting to see `to_string_view(ep.authn_type)` - since `rest_authn_type` doesn't have an `operator<<`.

Reply:
This will change when I remove the rest_authn_endpoint class anyways in favor of `authentication_method: <authn type>`

Reply:
I still need to make this change. It has cascading effects so I did everything else first.